src/term/mod.rs

```diff
@@ -1352,6 +1353,18 @@ impl ansi::Handler for Term {
     fn set_title(&mut self, title: &str) {
         if self.dynamic_title {
             self.next_title = Some(title.to_owned());
+
+            #[cfg(windows)]
+            {
+                // cmd.exe in winpty sends titles with blank space at the beginning
+                // because the title is not able to be passed to cmd.exe.
+                //
+                // The startswith check is necessary because other shells e.g. bash send a
+                // different title and don't need Alacritty appended.
+                if !tty::is_conpty() && self.next_title.as_ref().unwrap().starts_with(" ") {
```

Comment:
How does WinPTY reset the terminal title? If it just sends `""` and expects the terminal to reset the title to `Alacritty`, this obviously wouldn't work.

Reply:
Winpty always sends at least `' '` (a single space), as it incorrectly tells cmd.exe that that's the title (instead of `'Alacritty'`). That said, I've realised that what actually needs to be done is thus to replace the first space with `'Alacritty'` - otherwise we end up with an extra space compared to the conpty version. Have pushed a correction for that.